django/contrib/admin/views/autocomplete.py

```diff
@@ -90,7 +90,16 @@ def process_request(self, request):
                 type(model_admin).__qualname__
             )
 
+        # Resolve to_field_name:
+        # - Use attname for PKs.
+        # - Walk MTI hierarchy if necessary.
         to_field_name = getattr(source_field.remote_field, 'field_name', model_admin.model._meta.pk.name)
+        to_field = model_admin.model._meta.get_field(to_field_name)
+        if to_field.primary_key:
+            to_field_name = to_field.attname
+        while to_field.remote_field and to_field.remote_field.parent_link:
+            to_field_name = getattr(to_field.remote_field, 'field_name', to_field.remote_field.model._meta.pk.name)
+            to_field = to_field.remote_field.model._meta.get_field(to_field_name)
```

Comment:
@felixxm the `if to_field.primary_key` check is needed for the FK as PK case. 

Two questions: 

1. Can we tidy this block? 
2. Do we need the same kind of thing again inside the `while` loop? 
🤔

Reply:
In general, I think _it shouldn't be as hard as this_ (at least not exposed 😀)

Reply:
IMO there is enough to use `attname`, loop shouldn't be necessary. All tests work for me with (instead of L96-L102):
```python
        to_field_name = getattr(source_field.remote_field, 'field_name', remote_model._meta.pk.attname)
        to_field_name = model_admin.model._meta.get_field(to_field_name).attname
```

Reply:
Super — let me give that a run. Thanks @felixxm